mergin/editor.py

```diff
@@ -0,0 +1,94 @@
+from itertools import filterfalse
+
+from .utils import is_mergin_config, is_qgis_file, is_versioned_file
+
+EDITOR_ROLE_NAME = "editor"
+
+"""
+Determines whether a given file change should be disallowed based on the file path.
+
+Returns:
+    bool: True if the file change should be disallowed, False otherwise.
+"""
+_disallowed_added_changes = lambda change: is_qgis_file(change["path"]) or is_mergin_config(change["path"])
+"""
+Determines whether a given file change should be disallowed from being updated.
+
+The function checks the following conditions:
+- If the file path matches a QGIS file
+- If the file path matches a Mergin configuration file
+- If the file is versioned and the change does not have a diff
+
+Returns:
+    bool: True if the change should be disallowed, False otherwise.
+"""
+_disallowed_updated_changes = (
+    lambda change: is_qgis_file(change["path"])
+    or is_mergin_config(change["path"])
+    or (is_versioned_file(change["path"]) and change.get("diff") is None)
+)
+"""
+Determines whether a given file change should be disallowed from being removed.
+
+The function checks if the file path of the change matches any of the following conditions:
+- The file is a QGIS file (e.g. .qgs, .qgz)
+- The file is a Mergin configuration file (mergin-config.json)
+- The file is a versioned file (.gpkg, .sqlite)
+
+If any of these conditions are met, the change is considered disallowed from being removed.
+"""
+_disallowed_removed_changes = (
+    lambda change: is_qgis_file(change["path"]) or is_mergin_config(change["path"]) or is_versioned_file(change["path"])
+)
```

Comment:
A matter of taste, but the functions above would probably be a more natural fit for ordinary named functions (and we could have type hints with them as well)

Reply:
I added Callable type for these lambdas. Currently ```changes``` are just in dict type. 

I am waiting for proper type for changes dict handled in our system {"added": } - @varmar05  :100: